python/ray/tune/trial.py

```diff
@@ -88,7 +89,10 @@ def delete(checkpoint):
             checkpoint_path = checkpoint.value
             # Delete local copy, if any exists.
             if os.path.exists(checkpoint_path):
-                shutil.rmtree(checkpoint_path)
+                checkpoint_dir = TrainableUtil.find_checkpoint_dir(
```

Comment:
```suggestion
        try:
            checkpoint_dir = TrainableUtil.find_checkpoint_dir(checkpoint_path)
        except FileNotFoundError:
            logger.warning("Checkpoint not found during garbage collection.")
            return

```

Reply:
er also get rid of the `os.path.exists(checkpoint_path)` check since that check is made in `find_checkpoint_dir` already

Reply:
Is return necessary here?

Reply:
not if you move the rmtree call into the try

Reply:
If we don't check os.path.exists(checkpoint_path), I got a lot of warning message from line 97. If there're some legitimate cases that checkpoint_path does not exist, we might want to keep the check, instead of raising an exception.